src/NUnitConsole/nunit3-console/ConsoleRunner.cs

```diff
@@ -182,7 +183,19 @@ private int RunTests(TestPackage package, TestFilter filter)
                 {
                     throw new NUnitEngineException($"Error encountered in resolving output specification: {spec}", ex);
                 }
-
+                
+                try
+                {
+                    Directory.CreateDirectory(outputDirectory);
```

Comment:
I do like the idea of wrapping the exception. I think the wrapped string message gives the user a concise explanation of what's going wrong while the original error message can be used to debug if they need more information, . 

Also, I can see a few different exceptions being thrown by `CreateDirectory`, if the user doesn't have permission \(`UnauthorizedAccessException`\) or if the user inputs an invalid directory name \(`ArgumentNullException`, `DirectoryNotFoundException`, or `NotSupportedException`\).  

While taking a look at this, I also encountered another thing - an exception will be thrown by https://github.com/nunit/nunit-console/blob/f979cb049afa8cf809cdcd8e31a9ca54266a4c29/src/NUnitConsole/nunit3-console/ConsoleRunner.cs#L174 if the user path is too long \(`PathTooLongException`\) and it isn't currently wrapped.  

I suggest adding the line inside the try block before the `CreateDirectory`. This would make any issues arising from creating a new directory have a uniform error message.
https://github.com/nunit/nunit-console/blob/f979cb049afa8cf809cdcd8e31a9ca54266a4c29/src/NUnitConsole/nunit3-console/ConsoleRunner.cs#L187-L190

Reply:
The nested exception will give additional information, but only if the runner actually displays it. OTOH, the raw exception gets displayed with whatever message is there.

See also my other comment about the appropriateness of using an `NUnitEngineException` here. That's something the @nunit/engine-team will have to weigh in on.